### ISSUE TITLE
Following Quickstart Guide - Can't Deploy Kubeflow

### ISSUE THREAD

**Issue by MichaelJiangPlus** [kubeflow/kubeflow#1929]
I followed the instructions at https://www.kubeflow.org/docs/started/getting-started/ - and meet some question.

The start script went fine, but when I ran to `../scripts/kfctl.sh generate k8s`，Everything starts to be abnormal.He told me that there was a` runtime error: invalid memory address or nil pointer dereference
`

I tried a lot of methods and replaced a lot of versions of ks, but they didn't work.How can I locate and fix it?Or what should I do to get more detailed information to help me fix this problem

**This is my runtime script**
```
KUBEFLOW_SRC=kubeflow
KFAPP=kfapp
mkdir ${KUBEFLOW_SRC}
cd ${KUBEFLOW_SRC}
export KUBEFLOW_TAG=v0.3.1
curl https://raw.githubusercontent.com/kubeflow/kubeflow/${KUBEFLOW_TAG}/scripts/download.sh | bash

KFAPP=kfapp
scripts/kfctl.sh init ${KFAPP} --platform none
cd ${KFAPP}
../scripts/kfctl.sh generate k8s
../scripts/kfctl.sh apply k8s
```

**Here are some more detailed logs**
```
+ COMMAND=generate
+ WHAT=k8s
+ ENV_FILE=env.sh
+ SKIP_INIT_PROJECT=false
+ MIN_CLUSTER_VERSION=1.10.7-gke.2
+++ dirname ../scripts/kfctl.sh
++ cd ../scripts
++ pwd
+ DIR=/home/nvidia/Project/kubeflow/scripts
+ source /home/nvidia/Project/kubeflow/scripts/util.sh
+ source /home/nvidia/Project/kubeflow/scripts/gke/util.sh
++ set -xe
+ source /home/nvidia/Project/kubeflow/scripts/util-minikube.sh
++ RED='\033[0;31m'
++ GREEN='\033[0;32m'
++ YELLOW='\033[0;33m'
++ NC='\033[0m'
++ MOUNT_LOCAL=false
+ '[' generate == init ']'
+ source env.sh
++ PLATFORM=none
++ KUBEFLOW_REPO=/home/nvidia/Project/kubeflow
++ KUBEFLOW_VERSION=master
++ KUBEFLOW_KS_DIR=/home/nvidia/Project/kubeflow/kfapp/ks_app
++ KUBEFLOW_DOCKER_REGISTRY=
++ K8S_NAMESPACE=kubeflow
++ KUBEFLOW_CLOUD=null
+ '[' -z generate ']'
+ '[' -z k8s ']'
+ check_install ks
+ which ks
+ check_install kubectl
+ which kubectl
+ source env.sh
++ PLATFORM=none
++ KUBEFLOW_REPO=/home/nvidia/Project/kubeflow
++ KUBEFLOW_VERSION=master
++ KUBEFLOW_KS_DIR=/home/nvidia/Project/kubeflow/kfapp/ks_app
++ KUBEFLOW_DOCKER_REGISTRY=
++ K8S_NAMESPACE=kubeflow
++ KUBEFLOW_CLOUD=null
+ '[' generate == generate ']'
+ '[' k8s == platform ']'
+ '[' k8s == all ']'
+ '[' k8s == k8s ']'
+ createKsApp
+ pushd .
/home/nvidia/Project/kubeflow/kfapp /home/nvidia/Project/kubeflow/kfapp
++ dirname /home/nvidia/Project/kubeflow/kfapp/ks_app
+ cd /home/nvidia/Project/kubeflow/kfapp
++ basename /home/nvidia/Project/kubeflow/kfapp/ks_app
+ ks init ks_app
INFO Using context "kubernetes-admin@kubernetes" from kubeconfig file "/etc/kubernetes/admin.conf"
INFO Creating environment "default" with namespace "default", pointing to cluster at address "https://10.66.15.228:6443"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x456452]

goroutine 1 [running]:
github.com/ksonnet/ksonnet/vendor/github.com/ksonnet/ksonnet-lib/ksonnet-gen/ksonnet.parsePaths(0xc420198900, 0x6, 0x1, 0xc)
        /Users/bryan/go/src/github.com/ksonnet/ksonnet/vendor/github.com/ksonnet/ksonnet-lib/ksonnet-gen/ksonnet/paths.go:33 +0x1f1
github.com/ksonnet/ksonnet/vendor/github.com/ksonnet/ksonnet-lib/ksonnet-gen/ksonnet.NewCatalog(0xc420198900, 0xc4213bed90, 0x1, 0x1, 0x40, 0x0, 0x0)
        /Users/bryan/go/src/github.com/ksonnet/ksonnet/vendor/github.com/ksonnet/ksonnet-lib/ksonnet-gen/ksonnet/catalog.go:76 +0x1de
github.com/ksonnet/ksonnet/vendor/github.com/ksonnet/ksonnet-lib/ksonnet-gen/ksonnet.GenerateLib(0xc420460330, 0xe, 0x0, 0x0, 0x0)
        /Users/bryan/go/src/github.com/ksonnet/ksonnet/vendor/github.com/ksonnet/ksonnet-lib/ksonnet-gen/ksonnet/ksonnet.go:25 +0xce
github.com/ksonnet/ksonnet/pkg/util/kslib.astBuilder(0xc420460330, 0xe, 0xc420d66000, 0x3f0dc0, 0x3ffe00, 0x0, 0x0, 0x0)
        /Users/bryan/go/src/github.com/ksonnet/ksonnet/pkg/util/kslib/ksonnet.go:87 +0x40
github.com/ksonnet/ksonnet/pkg/util/kslib.Ksonnet(0xc420d66000, 0x3f0dc0, 0x3ffe00, 0x0, 0x0, 0x0)
        /Users/bryan/go/src/github.com/ksonnet/ksonnet/pkg/util/kslib/ksonnet.go:83 +0x3fd
github.com/ksonnet/ksonnet/pkg/lib.(*Manager).GenerateLibData(0xc42079a2c0, 0x1, 0x1dc69a0, 0x2a15f78)
        /Users/bryan/go/src/github.com/ksonnet/ksonnet/pkg/lib/lib.go:83 +0x8a
github.com/ksonnet/ksonnet/pkg/app.updateLibData(0x1dc69a0, 0x2a15f78, 0xc4214d67b0, 0xf, 0xc42079c240, 0x2e, 0x1, 0xc421567288, 0x43b4ab, 0xc4208007a8, ...)
        /Users/bryan/go/src/github.com/ksonnet/ksonnet/pkg/app/app.go:128 +0xba
github.com/ksonnet/ksonnet/pkg/app.(*App010).AddEnvironment(0xc42000c3c0, 0x187146a, 0x7, 0xc4214d67b0, 0xf, 0xc42151ae40, 0x0, 0x1a4, 0x0)
        /Users/bryan/go/src/github.com/ksonnet/ksonnet/pkg/app/app010.go:62 +0x427
github.com/ksonnet/ksonnet/pkg/env.(*creator).Create(0xc4201639e0, 0xc4201639e0, 0x200)
        /Users/bryan/go/src/github.com/ksonnet/ksonnet/pkg/env/create.go:117 +0x979
github.com/ksonnet/ksonnet/pkg/env.Create(0x1dcb1a0, 0xc42000c3c0, 0xc42042ac00, 0x19, 0x187146a, 0x7, 0x187146a, 0x7, 0xc4214d67b0, 0xf, ...)
        /Users/bryan/go/src/github.com/ksonnet/ksonnet/pkg/env/create.go:42 +0x157
github.com/ksonnet/ksonnet/pkg/appinit.(*initApp).Run(0xc4213bf980, 0x7f95c56b46c8, 0x0)
        /Users/bryan/go/src/github.com/ksonnet/ksonnet/pkg/appinit/init.go:88 +0x452
github.com/ksonnet/ksonnet/pkg/appinit.Init(0x1dc69a0, 0x2a15f78, 0x7ffe6b8357e2, 0x6, 0xc4207d6e10, 0x2a, 0x0, 0x0, 0xc4214d67b0, 0xf, ...)
        /Users/bryan/go/src/github.com/ksonnet/ksonnet/pkg/appinit/init.go:33 +0x14a
github.com/ksonnet/ksonnet/pkg/actions.(*Init).Run(0xc420163560, 0xc420163560, 0x0)
        /Users/bryan/go/src/github.com/ksonnet/ksonnet/pkg/actions/init.go:106 +0x121
github.com/ksonnet/ksonnet/pkg/actions.RunInit(0xc420d22630, 0xc4208b9b00, 0xe)
        /Users/bryan/go/src/github.com/ksonnet/ksonnet/pkg/actions/init.go:37 +0x5c
github.com/ksonnet/ksonnet/pkg/clicmd.runAction(0xe, 0xc420d22630, 0x188a651, 0x17)
        /Users/bryan/go/src/github.com/ksonnet/ksonnet/pkg/clicmd/actions.go:113 +0x66
github.com/ksonnet/ksonnet/pkg/clicmd.glob..func17(0x29d5da0, 0xc4202c72d0, 0x1, 0x1, 0x0, 0x0)
        /Users/bryan/go/src/github.com/ksonnet/ksonnet/pkg/clicmd/init.go:107 +0x743
github.com/ksonnet/ksonnet/vendor/github.com/spf13/cobra.(*Command).execute(0x29d5da0, 0xc4202c72a0, 0x1, 0x1, 0x29d5da0, 0xc4202c72a0)
        /Users/bryan/go/src/github.com/ksonnet/ksonnet/vendor/github.com/spf13/cobra/command.go:753 +0x468
github.com/ksonnet/ksonnet/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x29d96a0, 0x13, 0x18b9332, 0x30)
        /Users/bryan/go/src/github.com/ksonnet/ksonnet/vendor/github.com/spf13/cobra/command.go:843 +0x30a
github.com/ksonnet/ksonnet/vendor/github.com/spf13/cobra.(*Command).Execute(0x29d96a0, 0xc4200c0058, 0x0)
        /Users/bryan/go/src/github.com/ksonnet/ksonnet/vendor/github.com/spf13/cobra/command.go:791 +0x2b
main.main()
        /Users/bryan/go/src/github.com/ksonnet/ksonnet/cmd/ks/main.go:35 +0x85
```

**This is my cluster configuration**
```
NAME                       STATUS     ROLES    AGE     VERSION
blackwalnutlabs-vgpu-001   Ready      <none>   3d1h    v1.10.5
blackwalnutlabs-vgpu-002   NotReady   <none>   3d1h    v1.10.5
blackwalnutlabs-vgpu-003   Ready      <none>   3d1h    v1.10.5
blackwalnutlabs-vgpu-004   Ready      <none>   2d10h   v1.10.5
vgpu-master                Ready      master   3d1h    v1.12.2

```

**This is my ks version**
```
ksonnet version: 0.11.0
jsonnet version: v0.10.0
client-go version:
```

**Comment by pdmack** [comment]
I saw this also on RHEL 7.5 and resorted to building ksonnet locally with a newer version of golang (>1.9). I think there are some breaking changes between 1.9 and 1.10 that may be sensitive to the underlying C runtime. YMMV.

**Comment by MichaelJiangPlus** [comment]
It works！Thank you very much.I tried to install the 1.11.2 version of go and compiled the source code of ks.After compiling it works! 😆 

This is my current version information.

```
# go version
go version go1.11.2 linux/amd64

# ks version
ksonnet version: dev-2018-11-13T03:40:36-0500
jsonnet version: v0.11.2
client-go version: kubernetes-1.10.4
```

This is the script I used to install go and compile ks, hope to help others who have the same problem.

```
# USER root

# install go
wget https://dl.google.com/go/go1.11.2.linux-amd64.tar.gz
tar -C /usr/local -xzf go1.11.2.linux-amd64.tar.gz
export PATH=$PATH:/usr/local/go/bin

# make ksonnet
GOPATH=/root/go/
go get github.com/ksonnet/ksonnet
cd $GOPATH/src/github.com/ksonnet/ksonnet
make install
```

**Comment by edwardspc** [comment]
Same problem, but solved with ksonnet v0.13.1 : )
